tests/ui_tools/test_boxes.py

```diff
@@ -1306,12 +1311,22 @@ def test__stream_box_autocomplete_with_spaces(
     @pytest.mark.parametrize(
         "text, matching_topics",
         [
-            ("", ["Topic 1", "This is a topic", "Hello there!"]),
-            ("Th", ["This is a topic"]),
+            ("", ["Topic 1", "This is a topic", "Hello there!", "He-llo there!"]),
+            ("Th", ["This is a topic", "Hello there!", "He-llo there!"]),
+            ("the", ["Hello there!", "He-llo there!"]),
+            ("hel", ["Hello there!"]),
+            ("he", ["Hello there!", "He-llo there!"]),
+            ("he-llo", ["He-llo there!"]),
+            ("llo", ["He-llo there!"]),
         ],
         ids=[
             "no_search_text",
             "single_word_search_text",
+            "dash_sign_other_words_test",
+            "dash_sign_splited_or_not_0_test",
+            "dash_sign_splited_or_not_1_test",
+            "dash_sign_in_search_test",
+            "dash_sign_space_test",
```

Comment:
I'm not exactly familiar with pytest, is there any document on how to use "case" or "param"(as shown in test_boxes.py,` from pytest import param as case`) to tidy up the ids? I did some search and couldn't find any. Or do you mean that I can do that myself by naming them in a more distinguished way?

Reply:
I imported as `case` since `param` is used everywhere in pytest :)

This allows a test case that would be one list item as a tuple `(value1, value2, value3)` mapping to one list item in the `ids`, eg. `"some_test_case"`, to be written instead as
```python3
case(value1, value2, value3, id="some_test_case"),
case(value1a, value2, value3, id="another_test"),
...
```

That maintains locality of the ids with each specific test case :)